force_wfmanager/left_side_pane/workflow_tree.py

```diff
@@ -288,11 +318,46 @@ def delete_data_source(self, ui_info, object):
 
     def edit_data_source(self, ui_info, object):
         # This is a live dialog, workaround for issue #58
-        object.model.edit_traits(kind="livemodal")
+        ModelEditDialog(model=object.model).edit_traits()
 
     def new_layer(self, ui_info, object):
         self.workflow_mv.add_execution_layer(ExecutionLayer())
 
     def delete_layer(self, ui_info, object):
         """ Delete an element from the workflow """
         self.workflow_mv.remove_execution_layer(object.model)
+
+    def mv_editable(self, modelview):
```

Comment:
this code has a lot in common with the other code in _get_current_model_editable. There's potential for factoring out the commonalities in a single function